### ISSUE TITLE
Support any image for Icons

### ISSUE THREAD

**Issue by AnthonyOliveri** [carbon-design-system/carbon-components-react#182]
`Icon` components are currently restricted to a limited set of pre-made icons from the [Icon library](http://carbondesignsystem.com/style/iconography/library). For the Bluemix developer console, we need to show icons for a variety of services, each with their own icon that can change in the future. Many of these icons do not exist in the icon library, and it would not be feasible to constantly update the library for every new icon we need to display. Ideally, we would supply an `.svg` as a prop to `<Icon>` instead of a `name`.

**Comment by hellobrian** [comment]
I totally agree - I'll try to bring this up with the internal Carbon team to get this work prioritized. 
Need to explore how to implement this configuration. 

Curious to know how you expect the `<Icon>` component to be declared and used. 
Are you expecting to give a path to a `.svg` file? 

```jsx
<Icon src="./path/to/my-icon.svg" />
```

Or did you have something else in mind?

**Comment by AnthonyOliveri** [comment]
Thanks Brian! Yes, that is exactly what I was thinking. We currently are implementing our own version of `Icon` that accepts a `.svg` from either a local path or an external link. It's a workaround that works on older versions of `carbon-components-react`, but isn't compatible with newer versions (particularly >=3.5.10).

**Comment by jzhang300** [comment]
Here is our current implementation:
* React Component: https://github.ibm.com/Watson/watson-console-ui/blob/9267c47f35a05a09d3eb94c6667102137b9fcf1b/src/shared/components/Icon.js
* SCSS: https://github.ibm.com/Watson/watson-console-ui/blob/9267c47f35a05a09d3eb94c6667102137b9fcf1b/src/client/scss/components/icon/_icon.scss

Ideally, there should be some kind of fusion of the Carbon Icon and the one here.

**Comment by joshblack** [comment]
Hi there! 👋 

Sorry this issue has gone stale, unfortunately the team has been changing so things have gone through the cracks that I know no one meant to have happen.

For someone taking a look at this for the first time, I'd love to understand the usage a little better. The main ask here seems to be to leverage any `*.svg` asset in the Icon component, is that correct?

From an offering perspective, this seems like something that should be handled through the build tools of the offering itself. Is something like https://github.com/boopathi/react-svg-loader not sufficient for this use-case?

**Comment by hellobrian** [comment]
Hey @joshblack, I had a meeting with @jzhang300 and @AnthonyOliveri before leaving and I suggested `react-svg-loader` as well. Apologies if that offline discussion didn't propagate to others who wanted to hear updates on this. If I remember correctly, we also talked about making icons checkable but the implementation I saw was not optimal for accessibility. 

With all that said, I would say this issue can be closed or reopened to direct people to `react-svg-loader` through docs or whatever is most effective to encourage that loader usage. 👍
